lib/parser.js

```diff
@@ -299,19 +301,47 @@ Parser.prototype._trimNodes = function (node) {
   }
 };
 
+Parser.prototype._initializeUserVariables = function(filePath, config) {
+  const userDefinedVariables = {};
+  const baseDirectoryForFile = calculateNewBaseUrl(filePath, config.rootPath, config.baseUrlMap).relative;
+  const definedVariablePath = path.resolve(baseDirectoryForFile, USER_VARIABLES_PATH);
+  let content;
+  try {
+    content = fs.readFileSync(definedVariablePath, 'utf8');
+  } catch (e) {
+    content = '';
+    this._onError(e);
```

Comment:
Must this file always exist?

<img width="570" src="https://user-images.githubusercontent.com/14091939/37103478-4d86f1b4-2265-11e8-83e7-1e075687cbe2.png">

Currently, the terminal is spammed with the above error messages. I'd rather create a blank file after `_onError`.

Reply:
How about creating the file in markbind cli or checking if the file does not exist and warn once in cli (then remove this error check)?

Reply:
Sure.

> I think I might still need to make some changes
> * When a website includes a file X from a sub-site, we should use the sub-site's `variables.md` instead of the  website.
> * Can try to read the website and sub-site's `variables.md` once per Parser instead of on every include

You can do it similar to [`Site.prototype.collectBaseUrl`](https://github.com/MarkBind/markbind-cli/blob/8a0871d96f5ab12fb42afe352dd8b53691d04cd1/lib/Site.js#L173-L186).
Then pass `userDefinedVariables` in [`markbinder.includeFile`](https://github.com/MarkBind/markbind-cli/blob/6b04530a561b28015450beaf2af6decb1c4d4695/lib/Page.js#L81) as part of the [`config`](https://github.com/markbind/markbind/blob/215cfca4ee546bb32c95034b2c2f18d7cef3d703/lib/parser.js#L302) parameter.